From 8c5a5160c6783735b9eb7593ce614dea31ede56e Mon Sep 17 00:00:00 2001
From: Khoi Nguyen <khoi.nguyen.aj@renesas.com>
Date: Wed, 29 Jul 2020 13:56:51 +0700
Subject: [PATCH 322/355] arm64: dts: renesas: r8a774a3: Add VCP4 device nodes

Adds VCP4 device nodes (fcpcs, ivdp1c, vcplf, vdpb)
 used by UVCS driver on RZ/G2M v3.0 (a.k.a R8A774A3 SoC)

Signed-off-by: Khoi Nguyen <khoi.nguyen.aj@renesas.com>
---
 arch/arm64/boot/dts/renesas/r8a774a3.dtsi | 34 +++++++++++++++++++++++++++----
 1 file changed, 30 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a774a3.dtsi b/arch/arm64/boot/dts/renesas/r8a774a3.dtsi
index 5ebb34b..75edff3 100644
--- a/arch/arm64/boot/dts/renesas/r8a774a3.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a774a3.dtsi
@@ -2160,23 +2160,49 @@
 		};
 
 		fcpcs: vcp4@fe90f000 {
+			compatible = "renesas,vcp4-fcpcs";
 			reg = <0 0xfe90f000 0 0x200>;
-			/* placeholder */
+			clocks = <&cpg CPG_MOD 619>;
+			power-domains = <&sysc R8A774A3_PD_A2VC1>;
+			resets = <&cpg 619>;
+			renesas,#ch = <0>;
 		};
 
 		ivdp1c: vcp4@fe8d0000 {
+			compatible = "renesas,vcp4-ivdp1c";
 			reg = <0 0xfe8d0000 0 0x200>, <0 0xfe8d0200 0 0x200>;
-			/* placeholder */
+			interrupts = <GIC_SPI 380 IRQ_TYPE_LEVEL_HIGH>,
+					<GIC_SPI 381 IRQ_TYPE_LEVEL_HIGH>,
+					<GIC_SPI 223 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 130>;
+			power-domains = <&sysc R8A774A3_PD_A2VC1>;
+			resets = <&cpg 130>;
+			renesas,#ch = <2>;
+			renesas,#fcp_ch = <1>;
 		};
 
 		vcplf: vcp4@fe910000 {
+			compatible = "renesas,vcp4-vcplf";
 			reg = <0 0xfe910000 0 0x200>, <0 0xfe910200 0 0x200>;
-			/* placeholder */
+			interrupts = <GIC_SPI 260 IRQ_TYPE_LEVEL_HIGH>,
+					<GIC_SPI 261 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 130>;
+			power-domains = <&sysc R8A774A3_PD_A2VC1>;
+			resets = <&cpg 130>;
+			renesas,#ch = <0>;
+			renesas,#fcp_ch = <0>;
 		};
 
 		vdpb: vcp4@fe900000 {
+			compatible = "renesas,vcp4-vdpb";
 			reg = <0 0xfe900000 0 0x200>, <0 0xfe900200 0 0x200>;
-			/* placeholder */
+			interrupts = <GIC_SPI 240 IRQ_TYPE_LEVEL_HIGH>,
+					<GIC_SPI 241 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 131>;
+			power-domains = <&sysc R8A774A3_PD_A2VC1>;
+			resets = <&cpg 131>;
+			renesas,#ch = <1>;
+			renesas,#fcp_ch = <0>;
 		};
 
 		fdp0: fdpm@fe940000 {
-- 
2.7.4

